### DISCUSSION TITLE
2009/10/27/%E5%A5%B3%E5%84%BF%E7%9A%84%E5%90%8D%E5%AD%97

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#284] [Announcements]
# 2009/10/27/%E5%A5%B3%E5%84%BF%E7%9A%84%E5%90%8D%E5%AD%97

虽然我的下一代影都还没有呢，但我未雨绸缪，已经开始为他们想名字了。由于干我这行的倾向于生女儿，目前只考虑了女儿的名字。

https://blog.qizhen.xyz/2009/10/27/%E5%A5%B3%E5%84%BF%E7%9A%84%E5%90%8D%E5%AD%97

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Sophie
2010年05月9日 20:28 编辑
不懂labview, 但是真的觉得你很敬业，佩服！

回复
Tian
2010年01月8日 13:45 编辑
思奇满好的，我又想去青岛喝啤酒吃海鲜了

回复
arrowpig
2009年11月8日 08:55 编辑
恩仪 很好听

回复
心花怒放
2009年11月7日 17:10 编辑
说到给小孩起名，我过了年也有小孩了，不过名字已经起好了，起了一个男女通用的，呵呵。本来，我已经为小男孩起好了，可是老婆一时也不知道该给女孩起什么名字。后来，我老婆终于起了一个，宇，结果我我母亲正好过来，开口就起了一个，平，平平安安，我们觉得也不错，就基本定下了。因为我的姓写起来很复杂，所以，名字就起一个简单一点的，也不错。

回复
Qizhen
2009年11月6日 23:16 编辑
函数传递的参数无论什么数据类型都是存放在一块内存里的，所以有时可以用其它类型代替一下。对于tBCTW_DLDevInfo* 这个数据类型，你可以试着用一个有3076个元素u8数组来读一下。不过，LabVIEW毕竟不能支持所有复杂数据类型。在遇到类似的有复杂数据类型参数的情况下，我还是建议你自己再写一个DLL，把原来DLL函数中的复杂数据类型拆解成简单数据类型。然后在LabVIEW中使用你自己做的DLL。

回复
心花怒放
2009年11月6日 17:22 编辑
如果您需要调试的话，我可以把Dll和头文件发给您。

回复
心花怒放
2009年11月6日 17:21 编辑
实在不好意思啊，阮老师，刚才使用Opera发评论，没有成功，只好在这里再发一次了，请您指导下，谢谢了。

回复
心花怒放
2009年11月6日 17:20 编辑
阮老师，您好。对于“C 语言的结构中如果还嵌套了数组，是不能直接对应于LabVIEW中嵌套了数组的Cluster的。在LabVIEW中，只能把数组的元素都拆开来放在Cluster中。”我这里有一个用C++写的库，函数原型是BTCW_FUNC_DLL tBTCWStatus BTCW_FUNC_TYPE BTCW_DL_GetDevInfo(tBTCWHdl Hdl, unsigned __int16 index, tBCTW_DLDevInfo *pDLDevInfo); 关键看tBCTW_DLDevInfo，这是一个结构体，定义为typedef struct{ unsigned __int8 WriteVerifyMode; // tBTCW_DLWriteVerifyMode unsigned __int8 SectorEraseMode; // tBTCW_DLSectorEraseMode unsigned __int8 ConfigFixedHeader; // To decide if include fix header unsigned __int8 Reserved; unsigned __int8 MinidriverPath[BTCW_DL_DEV_PATH]; unsigned __int8 FirmwarePath[BTCW_DL_DEV_PATH]; unsigned __int8 ConfigPath[BTCW_DL_DEV_PATH];}tBCTW_DLDevInfo;而BTCW_DL_DEV_PATH预定义为#define BTCW_DL_DEV_PATH 1024。如果在导入的LabVIEW簇中不能直接使用数组，那这个簇里得有3000多个元素啊！！！其实，如果元素个数少的话，使用簇与数组的转换函数很容易处理。但是转换函数的限制是256，根本无法转换大数组啊。这个问题是我在使用导入共享库功能的时候发现的。当时，LabVIEW转换到这个函数时，像死机了一样，我等了一上午也没有转换过去，后来一看，估计是数组太大的原因。想请教一下阮老师有没有什么好的办法。先谢谢您了。另：还有一个比这个更头痛的函数：BTCW_FUNC_DLL tBTCWStatus BTCW_FUNC_TYPE BTCW_DL_StartDownload(tBTCWHdl Hdl, tBTCW_PortInfo *pBTCWPortInfo, tBTCW_Download *pDownload);tBTCW_PortInfo比较简单，tBTCW_Download定义是，typedef struct{ __int16 BTPIndex; unsigned __int8 UpdateWithDevInfo; // if UpdatwWithDevInfo is false, will use the values in BTP. Otherwise, use DLDevInfo tBCTW_DLDevInfo DLDevInfo; unsigned __int8 UpdateWithBDADDR; // if UpdateWithBDADDR is false, will use 0xFFFFFFFFFFFF as BDADDR unsigned __int8 SectorEraseTimeout; // if chip erase, then we need to wait time-out for erase command tBTCW_BDADDR BDADDR; tBTCW_DownloadCallBack pCallBack;}tBTCW_Download;这个就是结构里面有结构体了，唉。

回复
BO
2009年10月31日 07:32 编辑
思鲁会让我想起斯巴鲁。恩仪这名字不错，用了～

回复
Shi
2009年10月29日 17:10 编辑
说实话，阮恩仪，还真不错耶~顶一个

回复
佳
2009年10月29日 00:11 编辑
原来生活中还是挺有情趣的~~~

回复
Jucao
2009年10月28日 17:21 编辑
叫思齐的人是很多，见贤思齐嘛。毛泽东的儿媳就叫刘思齐。

回复
demi
2009年10月28日 12:11 编辑
阮思鲁也不错啊，又中性，又没有重名。而且鲁国首都是曲阜，还是孔子故乡。思齐主要是因为“见贤思齐”，所以叫的人比较多。

回复
帆
2009年10月28日 10:29 编辑
工作10年了也没有要孩子。。。算晚么？

回复
帆
2009年10月28日 10:24 编辑
干我这行的倾向于生女儿，这个有倾向性？

回复
shanghai_kiss
2009年10月28日 09:01 编辑
你的姓已经很特别了

回复
ruirui
2009年10月27日 22:18 编辑
我觉得ruan 和 en 有点拗口，而且也许孩子会被人起外号 阮阿姨。。。别打我啊，忠言逆耳

回复
永青
2009年10月27日 20:06 编辑
真敬业啊。。。

回复
